viewer/packages/sector-parser/src/GltfSectorParser.ts

```diff
@@ -0,0 +1,165 @@
+      const instancedBufferAttribute = new THREE.InstancedBufferAttribute(
+        accessor.getArray()!,
+        accessor.getElementSize()
+      );
+
+      bufferGeometry.setAttribute(`a${attributeName}`, instancedBufferAttribute);
+    });
+  }
+
+  private setUniqueMeshAttributes(
+    primitive: Primitive,
+    bufferGeometry: THREE.BufferGeometry | THREE.InstancedBufferGeometry
+  ) {
+    const treeIndexAttribute = primitive.getAttribute('_treeIndex')!;
+    const colorAttribute = primitive.getAttribute('COLOR_0')!;
+
+    bufferGeometry.setAttribute(
+      'color',
+      new THREE.BufferAttribute(colorAttribute.getArray()!, colorAttribute.getElementSize())
+    );
+
+    bufferGeometry.setAttribute(
+      'a_treeIndex',
+      new THREE.BufferAttribute(treeIndexAttribute.getArray()!, treeIndexAttribute.getElementSize())
+    );
+  }
+
+  private setMeshAttributes(
+    primitive: Primitive,
+    bufferGeometry: THREE.BufferGeometry | THREE.InstancedBufferGeometry
+  ) {
+    const indicesAccessor = primitive.getIndices()!;
+    const positionAttribute = primitive.getAttribute('POSITION')!;
+
+    bufferGeometry.setAttribute(
+      'position',
+      new THREE.BufferAttribute(positionAttribute.getArray()!, positionAttribute.getElementSize())
+    );
+    bufferGeometry.setIndex(new THREE.BufferAttribute(indicesAccessor.getArray()!, indicesAccessor.getElementSize()));
+  }
+
+  private setPrimitiveTopology(primitiveCollectionName: RevealGeometryCollectionType, geometry: THREE.BufferGeometry) {
+    switch (primitiveCollectionName) {
+      case RevealGeometryCollectionType.BoxCollection:
+        setBoxGeometry(geometry);
+        break;
+      case RevealGeometryCollectionType.CircleCollection:
+        setQuadGeometry(geometry); // should use the position as normal
+        break;
+      case RevealGeometryCollectionType.ConeCollection:
+        setConeGeometry(geometry);
+        break;
+      case RevealGeometryCollectionType.EccentricConeCollection:
+        setConeGeometry(geometry);
+        break;
+      case RevealGeometryCollectionType.EllipsoidSegmentCollection:
+        setConeGeometry(geometry);
+        break;
+      case RevealGeometryCollectionType.GeneralCylinderCollection:
+        setConeGeometry(geometry);
+        break;
+      case RevealGeometryCollectionType.GeneralRingCollection:
+        setQuadGeometry(geometry, false);
+        break;
+      case RevealGeometryCollectionType.NutCollection:
+        setNutGeometry(geometry);
+        break;
+      case RevealGeometryCollectionType.QuadCollection:
+        setQuadGeometry(geometry);
+        break;
+      case RevealGeometryCollectionType.TrapeziumCollection:
+        setTrapeziumGeometry(geometry);
+        break;
+      case RevealGeometryCollectionType.TorusSegmentCollection:
+        setTorusGeometry(geometry);
+        break;
+      default:
+        throw new Error(`${RevealGeometryCollectionType[primitiveCollectionName]} is not supported`);
```

Comment:
Tip: `assertNever(primitiveCollectionName)` is nice to use hear as it will fail on compile if we add a new type that we forget to add a case for.